tasks/autoTranslationTask.coffee

```
@@ -0,0 +1,164 @@
+module.exports = (grunt) ->
+  grunt.registerMultiTask 'autoTranslate', 'Will call google translate for all translatable missing strings \n and then add their translations to the appropriate file', ->
+    done = @async()
+
+    CSON = require 'cson'
+    _ = require 'underscore'
+    Translate = require('@google-cloud/translate')
+
+    #Google Cloud Platform project ID
+    projectId = @data.projectId
+    #Google Cloud Platform Translation API key
+    apiKey = @data.apiKey || ""
+    #Instantiates a client
+    if apiKey != ""

```

A missing API key is fatal, if you ask me. We need it to make requests, right? Might as well fail fast.

---

If there's no apiKey specified the translation library checks for a local variable (which is actually googles recommended authentication method). More info on what I mean here: https://cloud.google.com/translate/docs/auth#application_default_credentials